packages/core/src/helpers/validates.ts

```diff
@@ -54,7 +54,7 @@ export const validatePasswordUsingOWASP = (testConfig: Partial<TestConfig> | nul
 export const validateDomain = (t: TFunction) =>
   Yup.string()
     .matches(
-      /^((?:(?:(?:\w[.\-+]?)*)\w)+)((?:(?:(?:\w[.\-+]?){0,62})\w)+)\.(\w{2,6})$/,
+      /(?=.{4,253}$)^((([A-Za-z0-9]{1,63})|([0-9]{1}))([\.]{1}|[\-]{1,})){1,}((?=.[a-zA-Z]{1,})[0-9a-zA-Z]{1,25}){1}/,
       t('validation.must-be-a-valid-domain', 'Must be a valid domain')
```

Comment:
these two examples don't pass the validation but it's an edge case with numbers in the TLD and still far better than what we have so OK.

sub.mkyong-i.k.k.k.d.lklkllnfo.c00e
xn--d1ai6ai.xn--p1ai